content/technotes/2019-12-19-urls-checking/index.md

```diff
@@ -0,0 +1,456 @@
+---
+slug: "urls-tidying"
+title: "We cleaned our website URLs with R!"
+date: 2019-12-19
+authors:
+  - Maëlle Salmon
+  - Scott Chamberlain
+  - Stefanie Butland
+categories:
+  - technotes
+topicid: 1307
+tags:
+  - Markdown
+  - RMarkdown
+  - xml2
+  - commonmark
+  - crul
+  - hugo
+output: 
+  md_document:
+    variant: markdown_github
+    preserve_yaml: true
+---
+
+Last year we reported on [the joy of using commonmark and xml2 to parse
+Markdown content, like the source of this website built with
+Hugo](/technotes/2018/09/05/commonmark/), in particular [to extract
+links](https://ropensci.org/technotes/2018/09/05/commonmark/#urls-parsing),
+at the time merely to count them. How about we go a bit further and use
+the same approach to find links to be fixed? In this tech note we shall
+report our experience using R to find broken/suboptimal links and fix
+them.
+
+What is a bad URL?
+------------------
+
+We tackled a few URL issues:
+
+-   We had used absolute links (using our domain name) instead of
+    relative links which is in particular bad when looking at a preview
+    after a change. `https://ropensci.org/blog/` should be `/blog/`.
+
+-   Some internal and external links were broken, but we obviously did
+    not know which ones.
+
+-   A few links were short links (`bit.ly/blabla`) whereas it’s best to
+    store the actual link because the short link could break too.
+
+-   Some links were http links, although the same https link might work.
+
+-   There were links to ropensci.github.io documentation websites that
+    can be replaced with links to [our brand-new docs
+    server](/technotes/2019/06/07/ropensci-docs/).
+
+Please read if you try this at home/
```

Comment:
i'm not sure i understand this section title?

Reply:
It is supposed to be a bad pun "try this at home" where home/ looks like an URL

Reply:
ok, cool

Reply:
Well maybe I should remove it?

Reply:
it's prob ok as is

Reply:
should that heading be larger?

Reply:
I don't think so, it's level 2.